hasura/graphql-engine
# Improve 500 Error Page (close #272)

### Description
This is a partial fix for the issues raised in #272.  It improves the appearance of the page.
It makes the 500 page conform to the same structure as the 404 page. 
The 500 page now displays all information currently delivered.

It does not address two closely related issues. I will file sepate issues for these:
 
- The information is still not that useful.
- Some path segments are interpreted as table/field names. When invalid, the server responds 404, but this is treated as 'an error' and the 500 handler is invoked.
- We have two ErrorBoundary components. ErrorBoundary is a standard React class. I suspect we should refactor both of them (into differently named components).
- Design of this page and the 404 page should be reviewed by graphic artist.
- In particular, the list of components could still use some graphic improvement

### Affected components 
- Console

### Related Issues



#272
#1340 
#1886
#1887
#1888

### Solution and Design
This is a first step towards a more comprehensive overhaul of this area of the codebase.
This is a simple cosmetic cleanup. Code generally follows the 404/PageNotFound.js structure.

### Steps to test and verify
1. Start console normally
2. browse to a table browse page
3. change the name of the table in the URL bar to gibberish and type enter
Observe that the resulting page is now at least somewhat informative and looks better

### Limitations, known bugs & workarounds

Because the underlying error is a 404, if you reload the page, you get the actual 404 page.
